php-checks/src/main/java/org/sonar/php/checks/NullDereferenceInConditionalCheck.java

```diff
@@ -0,0 +1,181 @@
+public class NullDereferenceInConditionalCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE_FORMAT =
+    "Either reverse the equality operator in the \"%s\" null test, or reverse the logical operator that follows it.";
+
+  private static final List<Kind> AND_KINDS = ImmutableList.of(Kind.CONDITIONAL_AND, Kind.ALTERNATIVE_CONDITIONAL_AND);
+  private static final List<Kind> OR_KINDS = ImmutableList.of(Kind.CONDITIONAL_OR, Kind.ALTERNATIVE_CONDITIONAL_OR);
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    ExpressionTree comparedWithNullExpression = getComparedWithNullExpression(tree.leftOperand(), tree.getKind());
+    if (comparedWithNullExpression != null) {
+      tree.rightOperand().accept(new NullExpressionUsageVisitor(comparedWithNullExpression, context(), this));
+    }
+
+    super.visitBinaryExpression(tree);
+  }
+
+  @Nullable
+  private static ExpressionTree getComparedWithNullExpression(ExpressionTree expression, Tree.Kind kind) {
+    ExpressionTree comparedExpression = getExpressionEqualNullWithAnd(expression, kind);
+
+    if (comparedExpression == null) {
+      comparedExpression = getExpressionNotEqualNullWithOr(expression, kind);
+
+      if (comparedExpression == null) {
+        comparedExpression = getExpressionComparedWithFunction(expression, kind);
+
+        if (comparedExpression == null) {
+          return null;
+        }
+      }
+    }
+
+    return comparedExpression;
+  }
+
+  @Nullable
+  private static ExpressionTree getExpressionEqualNullWithAnd(ExpressionTree tree, Kind kind) {
+    if (AND_KINDS.contains(kind)) {
+      return getComparedWithNull(tree, Tree.Kind.EQUAL_TO, Tree.Kind.STRICT_EQUAL_TO);
+    }
+
+    return null;
+  }
+
+  @Nullable
+  private static ExpressionTree getExpressionNotEqualNullWithOr(ExpressionTree tree, Kind kind) {
+    if (OR_KINDS.contains(kind)) {
+      return getComparedWithNull(tree, Tree.Kind.NOT_EQUAL_TO, Tree.Kind.STRICT_NOT_EQUAL_TO);
+    }
+
+    return null;
+  }
+
+  /**
+   *  e.g.      is_null($obj) && ...     or      !is_null($obj) || ...
+   */
+  @Nullable
+  private static ExpressionTree getExpressionComparedWithFunction(ExpressionTree expression, Kind kind) {
+    if (OR_KINDS.contains(kind) && expression.is(Kind.LOGICAL_COMPLEMENT) && ((UnaryExpressionTree) expression).expression().is(Kind.FUNCTION_CALL)) {
+      FunctionCallTree functionCall = (FunctionCallTree)((UnaryExpressionTree) expression).expression();
+      return retrieveArgumentFromIsNullCall(functionCall);
+    }
+
+    if (AND_KINDS.contains(kind) && expression.is(Kind.FUNCTION_CALL)) {
+      return retrieveArgumentFromIsNullCall((FunctionCallTree)expression);
+    }
+
+    return null;
+  }
+
+  @Nullable
+  private static ExpressionTree retrieveArgumentFromIsNullCall(FunctionCallTree functionCall) {
+    if (CheckUtils.asString(functionCall.callee()).equalsIgnoreCase("is_null") && functionCall.arguments().size() == 1) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "is_null" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1132)